eng/pipelines/templates/steps/install-dotnet.yml

```diff
@@ -0,0 +1,13 @@
+steps:
+  - task: UseDotNet@2
+    displayName: 'Use .NET Core SDK'
+    inputs:
+      useGlobalJson: true
+      performMultiLevelLookup: true
+  - task: UseDotNet@2
+    condition: ne(variables['Agent.OS'], 'Windows_NT') # Windows supports MultiLevelLookup and doesn't need explicit framework installation
```

Comment:
Is this new? How was it being handled before?

Reply:
Before we disabled the multi-level lookup and installed all versions of the runtime we need. This way we can only install what's not on the VM and let the dotnet.exe fallback to program files when required.

Reply:
https://github.com/Azure/azure-sdk-for-net/issues/17118